File: python3/float.py
# FOR EDUCATIONAL PURPOSES ONLY, DO NOT USE IN PRODUCTION
# Source code is provided as-is without warranty, and disclaiming liability for damages resulting from using the source code.
# Convert single precision float to 32 bits binary and back to single precsision float

import struct;

a = 3.14
print("Convert float:  {:f}".format(a));
bytes = struct.pack('f', a);
buf = "";
for i in range(4):
  buf += '{:02x} '.format(ord(bytes[i]));
print("To 32bit value: " + buf);
b = struct.unpack("f", bytes);
print("Back to float:  {:f}".format(b[0]));

# Result:
# Convert float:  3.140000
# To 32bit value: c3 f5 48 40
# Back to float:  3.140000

File: linux_c/float.c
// FOR EDUCATIONAL PURPOSES ONLY, DO NOT USE IN PRODUCTION
// Source code is provided as-is without warranty, and disclaiming liability for damages resulting from using the source code.
// Convert 4 bytes to single precision float using an union.

#include <stdio.h>
#include <stdint.h>
#include <string.h>

float toFlt(uint8_t *);

union{
  float fl;
  uint8_t bt[4];
}spf;

int main(){
  uint8_t f1[4] = {0x00, 0x58, 0x9c, 0x45};
  memcpy(spf.bt, f1, sizeof(spf.bt));
  printf("%.1f\n", spf.fl);
  printf("%.1f\n", toFlt(f1));
}

float toFlt(uint8_t *buf){
  memcpy(spf.bt, buf, sizeof(spf.bt));
  return spf.fl;
}
